### ISSUE TITLE
2.0.0-beta1 Init script startup fails silently when hitting java.lang.UnsupportedClassVersionError (unsupported JVM)

### ISSUE THREAD

**Issue by nellicus** [elastic/elasticsearch#13392]
OS info
```
elk@master-clone:/opt$ uname -a
Linux master-clone 3.2.0-4-amd64 #1 SMP Debian 3.2.68-1+deb7u3 x86_64 GNU/Linux
```

Java info
```
elk@master-clone:/opt$ java -version
java version "1.6.0_36"
OpenJDK Runtime Environment (IcedTea6 1.13.8) (6b36-1.13.8-1~deb7u1)
OpenJDK 64-Bit Server VM (build 23.25-b01, mixed mode)
```

Download and Install deb (probably affects other packages)
```
elk@master-clone:/opt$ wget https://download.elastic.co/elasticsearch/elasticsearch/elasticsearch-1.7.1.deb
--2015-09-08 13:31:03--  https://download.elastic.co/elasticsearch/elasticsearch/elasticsearch-1.7.1.deb
Resolving download.elastic.co (download.elastic.co)... 107.22.213.126, 107.22.237.122, 107.20.222.112, ...
Connecting to download.elastic.co (download.elastic.co)|107.22.213.126|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 27321880 (26M) [application/x-debian-package]
Saving to: `elasticsearch-1.7.1.deb'

100%[=================================================================================================================================================================================================================================================================================>] 27,321,880  4.43M/s   in 6.6s    

2015-09-08 13:31:11 (3.94 MB/s) - `elasticsearch-1.7.1.deb' saved [27321880/27321880]

elk@master-clone:/opt$ sudo dpkg -i elasticsearch-1.7.1.deb 
Selecting previously unselected package elasticsearch.
(Reading database ... 143837 files and directories currently installed.)
Unpacking elasticsearch (from elasticsearch-1.7.1.deb) ...
Creating elasticsearch group... OK
Creating elasticsearch user... OK
Setting up elasticsearch (1.7.1) ...
```

Launch service
```
elk@master-clone:/opt$ sudo /etc/init.d/elasticsearch start
[FAIL] Starting Elasticsearch Server: failed!
```

removing -b flag , from /etc/init.d/elasticsearch
```
start-stop-daemon --start -b --user "$ES_USER" -c "$ES_USER" --pidfile "$PID_FILE" --exec $DAEMON -- $DAEMON_OPTS
```

to

```
start-stop-daemon --start --user "$ES_USER" -c "$ES_USER" --pidfile "$PID_FILE" --exec $DAEMON -- $DAEMON_OPTS
		start-stop-daemon --stop --pidfile "$PID_FILE" \
```

to make this surface

```
elk@master-clone:/opt$ sudo /etc/init.d/elasticsearch start
[....] Starting Elasticsearch Server:Exception in thread "main" java.lang.UnsupportedClassVersionError: org/elasticsearch/bootstrap/Elasticsearch : Unsupported major.minor version 51.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:643)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:277)
	at java.net.URLClassLoader.access$000(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:212)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:205)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:323)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:294)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:268)
Could not find the main class: org.elasticsearch.bootstrap.Elasticsearch. Program will exit.
 failed!
```

This is really bad user experience and it would be very desirable at least to catch this output and present it to the user.

**Comment by nellicus** [comment]
will try this on 2.x

**Comment by nellicus** [comment]
still reproducible on 2.x I'm afraid

```
Last login: Tue Sep  8 11:12:34 2015 from antonios-macbook-air.local
elk@master-clone:~$ wget https://download.elasticsearch.org/elasticsearch/release/org/elasticsearch/distribution/deb/elasticsearch/2.0.0-beta1/elasticsearch-2.0.0-beta1.deb
--2015-09-09 17:31:08--  https://download.elasticsearch.org/elasticsearch/release/org/elasticsearch/distribution/deb/elasticsearch/2.0.0-beta1/elasticsearch-2.0.0-beta1.deb
Resolving download.elasticsearch.org (download.elasticsearch.org)... 107.21.118.106, 107.20.198.195, 184.72.232.248, ...
Connecting to download.elasticsearch.org (download.elasticsearch.org)|107.21.118.106|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 28431018 (27M) [application/x-debian-package]
Saving to: `elasticsearch-2.0.0-beta1.deb'

100%[=================================================================================================================================================================================================================================================================================>] 28,431,018  4.48M/s   in 9.1s    

2015-09-09 17:31:19 (2.99 MB/s) - `elasticsearch-2.0.0-beta1.deb' saved [28431018/28431018]

elk@master-clone:~$ sudo dpkg -i elasticsearch-2.0.0-beta1.deb 
[sudo] password for elk: 

Selecting previously unselected package elasticsearch.
(Reading database ... 143837 files and directories currently installed.)
Unpacking elasticsearch (from elasticsearch-2.0.0-beta1.deb) ...
Creating elasticsearch group... OK
Creating elasticsearch user... OK
Setting up elasticsearch (2.0.0~beta1) ...

elk@master-clone:~$ uname -a
Linux master-clone 3.2.0-4-amd64 #1 SMP Debian 3.2.68-1+deb7u3 x86_64 GNU/Linux
elk@master-clone:~$ cat /etc/debian_version 
7.9
elk@master-clone:~$ java -version
java version "1.6.0_36"
OpenJDK Runtime Environment (IcedTea6 1.13.8) (6b36-1.13.8-1~deb7u1)
OpenJDK 64-Bit Server VM (build 23.25-b01, mixed mode)
elk@master-clone:~$ time sudo /etc/init.d/elasticsearch start
[FAIL] Starting Elasticsearch Server: failed!

real	0m11.111s
user	0m0.004s
sys	0m0.012s
elk@master-clone:~$ sudo cp /etc/init.d/elasticsearch ~/
elk@master-clone:~$ egrep start-stop-daemon /etc/init.d/elasticsearch 
# Description:       Starts elasticsearch using start-stop-daemon
	start-stop-daemon -d $ES_HOME --start -b --user "$ES_USER" -c "$ES_USER" --pidfile "$PID_FILE" --exec $DAEMON -- $DAEMON_OPTS
		start-stop-daemon --stop --pidfile "$PID_FILE" \
```

remove background flag
```
elk@master-clone:~$ sudo sed -i 's/\-b//g' /etc/init.d/elasticsearch 
```

```
elk@master-clone:~$ egrep start-stop-daemon /etc/init.d/elasticsearch 
# Description:       Starts elasticsearch using start-stop-daemon
	start-stop-daemon -d $ES_HOME --start  --user "$ES_USER" -c "$ES_USER" --pidfile "$PID_FILE" --exec $DAEMON -- $DAEMON_OPTS
		start-stop-daemon --stop --pidfile "$PID_FILE" \
elk@master-clone:~$ time sudo /etc/init.d/elasticsearch start
[....] Starting Elasticsearch Server:Exception in thread "main" java.lang.UnsupportedClassVersionError: org/elasticsearch/bootstrap/Elasticsearch : Unsupported major.minor version 51.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:643)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:277)
	at java.net.URLClassLoader.access$000(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:212)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:205)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:323)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:294)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:268)
Could not find the main class: org.elasticsearch.bootstrap.Elasticsearch. Program will exit.
 failed!

real	0m11.101s
user	0m0.004s
sys	0m0.008s
```

**Comment by nik9000** [comment]
I've dropped the discuss label because I'm pretty sure this is worth fixing. I'll have a go at it sometime in the next few days if no one else wants it. I'm claiming it now so I don't forget about it but if anyone wants it chime in. I don't enjoy working on packaging issues but I enjoy having them solved....

**Comment by rmuir** [comment]
Its too bad `java` launcher is so broken, since it has a `-version:xxx` with fancy support for lists, wildcards, ranges, etc, to specify what is needed.

But it does not work at all.

**Comment by nik9000** [comment]
OK! I'm going to have a look at this.

I noticed that in systemd we've got elasticsearch setup as a `simple` service (the default) but if we switched to a `notify` service then we'd be able to send notifications back to systemd over a socket that tells it how Elasticsearch is doing. Neat stuff. Minimally I'd hope that we could send it a notification once the node is started. We could even send it information about where we are in the startup process.... Neat! Anyway, [man sd_notify](http://www.dsm.fordham.edu/cgi-bin/man-cgi.pl?topic=sd_notify&ampsect=3) has good docs for this. sd_notify doesn't have a Java library so that'd be trouble there.

Now I'll investigate the init script side....
